docs/modules/ROOT/pages/mgmt_services/route_manager.adoc

```diff
@@ -0,0 +1,1227 @@
+= Route Manager Design Specification
+Chun-Jen Chung <cchung@futurewei.com>, Liguang Xie <lxie@futurewei.com>
+v0.1, 2020-08-17
+:toc: right
+:imagesdir: ../../images
+
+== Overview
+
+The Route Manager is an Alcor microservice which manages the route tables and route entries for all VPCs and Subnets.
+It also provides router-related functionalities to compatible with OpenStack Neutron's router APIs.
+Its responsibilities include but are not limited to creating, updating, deleting, and searching routers, route tables,
+and route entries resource for a Network/VPC/Subnet.
+It interacts with VPC Manager, Subnet Manager, Port Manager, and Data Plane Manager to provide route information for them.
+
+== Service Requirements
+
+[arabic]
+. Availability Zone and Subnet-based route table.
+. Default route tables for VPC and subnet.
+. Default route to Internet Gateway in a public subnet route table.
+. Default route to NAT Gateway in a private subnet route table.
+. Router resource managed by an owner, Neutron or VPC.
+. Route table resource managed by an owner, VPC or Subnet.
+. Route entry should include destination, nexthop, target, and priority.
+. Warning message should be raised if same route entry with different priority has been inserted by users.
```

Comment:
One more requirement:
- Warning message should be raised when adding an invalid routing rule

Reply:
updated